adr/2025_03_10-suggestions/README.md

```diff
@@ -0,0 +1,178 @@
+# Versioning & Suggestions
+
+Due to the similarity between document versioning, red lining, track changes and suggestions, it is worth seeing an overview of all of their requirements to make sure we consider solutions that resolve multiple of them.
+
+- A diff between two versions of a document, is essentially a set of suggestions that takes your previous document, and transforms it into the more recent document.
+- Red-lining and suggestions are the same, red-lining just allows you to attach metadata to the suggestion, like a comment of why the change was made.
+- Track changes, is entering a mode where any edit you make, becomes a suggestion to the document.
+
+## Versioning Overview
+
+Versioning has already been implemented for Y.js & y-prosemirror, but to be explicit about the sort of thing we are looking for it is good to explicitly state the feature set versioning should have:
+
+- A user can take _snapshots_ of the current document state, and store that in a separate data store
+- These snapshots can be viewed independently at any point in time later
+- These snapshots can be _compared_ with each other
+  - This allows the user to see what changed between the two snapshots
+  - Changes should show:
+    - The content that was changed (i.e. additions or deletions)
+    - Updates to content (i.e. changing an attribute)
+    - Which user identifiers modified the content
+    - Ideally, a timestamp of when the content was modified
```

Comment:
Flesh out timestamp _means_ and at what granularity (typed first keystroke, or when snapshot was taken, or when user assumed a clientID)

Reply:
Can be derived from snapshotted version?